docs/pages/machine-id/reference/configuration.mdx

```diff
@@ -0,0 +1,94 @@
+---
+title: Machine ID Configuration Reference
+description: Configuration reference for Teleport Machine ID.
+---
+
+```yaml
+# Debug enables verbose logging to stderr.
+debug: true
+
+# Address of the Teleport Auth Server (on-prem installs) or Proxy Server
+# (Cloud installs).
+auth_server: "auth.example.com:3025" # or "example.teleport.sh:443" for Teleport Cloud
+
+# TTL of short-lived machine certificates.
+certificate_ttl: "5m"
+
+# Interval at which short-lived certificates are renewed; must be less than
+# the certificate TTL.
+renewal_interval: "1m"
+
+# If set, quit after the first renewal.
+oneshot: true
+
+# Onboarding values are only used on first connect.
+onboarding:
+    # Cluster join method. Can be "token" or "iam".
+    join_method: "token"
+    # Token used to join the cluster.
+    token: "00000000000000000000000000000000"
+    # CA Path used to validate the identity of the Teleport Auth Server on first connect.
+    ca_path: "/path/to/ca.pem"
+    # CA Pins used to validate the identity of the Teleport Auth Server on first connect.
+    ca_pins:
+    - "sha256:1111111111111111111111111111111111111111111111111111111111111111"
+    - "sha256:2222222222222222222222222222222222222222222222222222222222222222"
+
+# Storage defines where Machine ID internal data is stored.
+storage:
+  # Directory to store internal bot data. Access to this directory should be
+  # limited.
+  directory: /var/lib/teleport/bot
+
+  # Alternatively, internal data can be stored in memory. "directory" and
+  # "memory" are mutually exclusive.
+  memory: true
```

Comment:
Any concern about people copy-pasting this file, changing the tokens/endoints, and not seeing this comment and hitting an error on first start?

Maybe we should comment out `# memory: true` so that the example is still there but the config itself is valid.

Reply:
@timothyb89 mentioned the same, but everywhere else in our reference documentation we document all the options available without commenting out any blocks, even if they are mutually exclusive.

https://goteleport.com/docs/setup/reference/config/